test/runtests.jl

```diff
@@ -198,6 +198,17 @@ end
     @test m.a === m_inner_init
 end
 
+struct A{X, Y}
+    x::X
+    y::Y
+end
+
+@testset "type change during @set (default constructor_of)" begin
```

Comment:
Is this the test you mentioned in https://github.com/jw3126/Setfield.jl/pull/19#issuecomment-371087033 ?

Reply:
Yes, this looks good.